woocommerce/woocommerce-android
# [Login] Allow entering a site address manually when encountring the no-stores erro

Closes: #7215 


### Description
This PR adds ability to enter a site address manually when the user encounters the no-stores error, after entering the site address, we will run a site discovery that reuses the same logic from the Login library, in order to do so, I opted for creating a separate Activity that implements `LoginListener`, this allows us to reuse the Login fragments, but gives us more flexibility in adapting the flow to our usecase.

### Testing instructions
For all the scenarios below, please start with a WordPress.com account that's not connected to any site, then when the no-stores error is shown, click on "Enter a site address"

##### Site without Jetpack
1. Enter the address of a site that doesn't have Jetpack.
2. Confirm that a screen that explains the issue is shown, with a CTA to install Jetpack.
3. Try installing Jetpack then connecting it to your WordPress.com account.
4. Confirm that in the WebView you are automatically signed in to your WordPress.com account.
5. Confirm that after installation, you can continue login.

##### Jetpack sites and WPCom sites
For all of these cases, you should land on the "account mismatch" error.

##### Non-wordpress site
Confirm that the app shows the appropriate error.

### Screenshots
<img src="https://user-images.githubusercontent.com/1657201/185133347-a242ba12-0287-4702-94e6-a9e24080dc6a.png" width=360/>
